library/src/main/java/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -29,44 +28,50 @@ class MessageListViewModel(private val cid: String,
         loading.value = State.Loading
 
         val result = domain.useCases.watchChannel.invoke(cid, MESSAGES_LIMIT).execute()
-        channelController = result.data()
+        val channelController = result.data()
         channel = channelController.toChannel()
         currentUser = domain.currentUser
 
-        val resultState = MessageListItemLiveData(
+        messageListData = MessageListItemLiveData(
                 currentUser,
                 channelController.messages,
                 threadMessages,
                 channelController.typing,
                 channelController.reads
         )
-        .run {
-            map(this) { State.Result(it) }
-        }
 
         stateMerger.apply {
-            addSource(loading) {
-                value = it
-            }
-            addSource(resultState) {
-                value = it
-            }
+            addSource(loading) { value = it }
+            addSource(messageListData) { value = State.Result(it) }
         }
     }
 
     fun onEvent(event: Event) {
         when (event) {
```

Comment:
Are all event types handled?
What about adding a new one? This method won't handle it.
If you don't want to return anything on the function and not use the return function body, I created an extension (`.exhaustive`) that you can add at the end of the `when` sentence and It will fail in compile-time in the future if there is any new type that is not handled here